test/index.js

```diff
@@ -1,118 +1,108 @@
+				this.disk = new lkl.disk.FileDisk(DISK_PATH);
+			});
+
+			it('should mount vfat', function() {
+				return lkl.mountAsync(this.disk, {readOnly: true, filesystem: 'vfat', partition: 1})
+				.then(lkl.umountAsync)
+			});
+
+			it('should mount ext2', function() {
+				return lkl.mountAsync(this.disk, {readOnly: true, filesystem: 'ext2', partition: 2})
+				.then(lkl.umountAsync)
+			});
+
+			it('should mount ext4', function() {
+				return lkl.mountAsync(this.disk, {readOnly: true, filesystem: 'ext4', partition: 3})
+				.then(lkl.umountAsync)
 			});
-		}
-		// Wait for all writes to complete
-		async.parallel(calls, done);
+
+			it('should mount btrfs', function() {
+				return lkl.mountAsync(this.disk, {readOnly: true, filesystem: 'btrfs', partition: 5})
+				.then(lkl.umountAsync)
+			});
+
+			it('should mount xfs', function() {
+				return lkl.mountAsync(this.disk, {readOnly: true, filesystem: 'xfs', partition: 6})
+				.then(lkl.umountAsync)
+			});
+		});
 	});
 
 	describe('fs', function() {
-		describe('.access()', function() {
-			let removeFile = function(file) {
-				try {
-					lkl.fs.unlinkSync(file);
-				} catch (err) {
-					// Ignore error
-				}
-			};
+		before(function(done) {
+			self = this;
+
+			fs.createReadStream(RAW_FS_PATH)
+			.pipe(fs.createWriteStream(TMP_RAW_FS_PATH))
+			.on('close', function() {
+				const disk = new lkl.disk.FileDisk(TMP_RAW_FS_PATH);
+
+				lkl.mountAsync(disk, {filesystem: 'ext4'})
+				.then(function (mountpoint) {
+					self.mountpoint = mountpoint;
+				}).nodeify(done);
+			});
+		});
 
+		after(function() {
+			return lkl.umountAsync(this.mountpoint)
+			.then(function() {
+				return fs.unlinkAsync(TMP_RAW_FS_PATH)
+			});
+		});
+
+		describe('.access()', function() {
 			let createFileWithPerms = function(file, mode) {
-				removeFile(file);
-				lkl.fs.writeFileSync(file, '');
-				lkl.fs.chmodSync(file, mode);
+				// FIXME: remove the catch clause once unlink gets implemented
+				return lkl.fs.unlinkAsync(file).catch(function() {})
+				.then(function() {
+					return lkl.fs.writeFileAsync(file, '');
+				}).then(function() {
+					return lkl.fs.chmodAsync(file, mode);
+				});
 			};
 
 			before(function() {
-				this.doesNotExist = path.join(mountpoint, '__this_should_not_exist');
-				this.readOnlyFile = path.join(mountpoint, 'read_only_file');
-				this.readWriteFile = path.join(mountpoint, 'read_write_file');
-
-				createFileWithPerms(this.readOnlyFile, 0o444);
-				createFileWithPerms(this.readWriteFile, 0o666);
-			});
-
-			after(function() {
-
+				this.doesNotExist = path.join(this.mountpoint, '__this_should_not_exist');
+				this.readOnlyFile = path.join(this.mountpoint, 'read_only_file');
+				this.readWriteFile = path.join(this.mountpoint, 'read_write_file');
+
+				return Promise.all([
+					createFileWithPerms(this.readOnlyFile, 0o444),
+					createFileWithPerms(this.readWriteFile, 0o666)
+				]);
 			});
 
 			it('non existent file', function(done) {
```

Comment:
Sure, if mounting works it means that request is called.
What I had in mind was a code sample (that can also be a test) that shows how to intercept reads and writes.